File: array/twoSum.java
package array;

public class twoSum {

    public static int[] twosum(int[] arr, int target)
    {
        int[] newArr = new int[2];
        for (int i = 0; i<arr.length; i++)
        {
            for (int j = 1; j<arr.length ; j++)
            {
                if (arr[i] +  arr[j] == target && i!=j)
                {
                    newArr[0] = i;
                    newArr[1] = j;
                    return newArr;
                }
            }
        }

        return arr;
    }
;
    public static void main(String[] args) {
        int[] arr = {2,5,5,11};

        int[] array = twosum(arr,10);
        for (int i = 0; i<array.length; i++)
        {
            System.out.print(array[i] + " ");
        }

    }
}

File: leetcode_questions/twosum.cpp
/*
time - O(n^2)*/

#include <bits/stdc++.h>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int> &nums, int target)
{
    int n = nums.size();
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            if (nums[i] + nums[j] == target)
            {
                return {i, j};
            }
        }
    }
    return {};
}

int main()
{
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;

    vector<int> ans = twoSum(nums, target);
    for (int i = 0; i < ans.size(); i++)
    {
        cout << ans[i] << " ";
    }

    return 0;
}